app/controllers/katello/api/v2/repositories_controller.rb

```diff
@@ -264,8 +264,14 @@ def repository_types
 
     api :PUT, "/repositories/:id/republish", N_("Forces a republish of the specified repository, regenerating metadata and symlinks on the filesystem.")
     param :id, :number, :desc => N_("Repository identifier"), :required => true
+    param :force, :bool, :desc => N_("Force metadata regeneration to proceed.  Dangerous when repositories use mirror on sync."), :required => true
     def republish
-      task = async_task(::Actions::Katello::Repository::MetadataGenerate, @repository, :force => true)
+      unless ::Foreman::Cast.to_bool(params[:force])
+        render_error('custom_error', :status => :unprocessable_entity,
```

Comment:
The pattern I've seen for raising controller errors has been more like `fail HttpErrors::BadRequest("why u do this?")`. Was there an advantage to going this route 'cause I think raising an error like BadRequest ultimately does what you've done here automatically!

Reply:
No reason, that looks better to me.